src/Command/Project/ProjectCreateCommand.php

```diff
@@ -247,23 +317,85 @@ protected function getFields()
             },
             'default' => in_array('development', $this->getAvailablePlans()) ? 'development' : null,
             'allowOther' => true,
-          ]),
-          'environments' => new Field('Environments', [
+          ]);
+          $fields['environments'] = new Field('Environments', [
             'optionName' => 'environments',
             'description' => 'The number of environments',
             'default' => 3,
             'validator' => function ($value) {
                 return is_numeric($value) && $value > 0 && $value < 50;
             },
-          ]),
-          'storage' => new Field('Storage', [
+          ]);
+          $fields['storage'] = new Field('Storage', [
             'description' => 'The amount of storage per environment, in GiB',
             'default' => 5,
             'validator' => function ($value) {
                 return is_numeric($value) && $value > 0 && $value < 1024;
             },
-          ]),
-        ];
+          ]);
+        return $fields;
+    }
+
+     /**
+     * Returns a list of ConsoleForm form fields for this command.
+     *
+     * @return Field[]
+     */
+    protected function getTemplateFields($url_provided)
+    {
+        $fields = [];
+        
+        if (!$url_provided) {
+            $fields['template_option'] = new OptionsField('Template Options', [
+                'optionName' => 'template_option',
+                'options' => [
+                    'Provide your own template url.',
+                    'Choose a template from the catalog.',
+                    'No template at this time.',
+                ],
+                'description' => 'Choose a template, provide a url or choose not to use one.',
+                'includeAsOption' => false,
+            ]);
+    
+            $fields['catalog_url'] = new OptionsField('Catalog', [
+            'optionName' => 'catalog_url',
+            'conditions' => [
+                'template_option' => [
+                    'Choose a template from the catalog.'
+                ],
+            ],
+            'description' => 'The template from which to create your project or your own blank project.',
+            'options' => $this->getAvailableCatalog(),
+            'asChoice' => FALSE,
+            'optionsCallback' => function () {
+                return $this->getAvailableCatalog(true);
+                },
+            ]);
+
+            $fields['template_url'] = new UrlField('Template URL', [
```

Comment:
Should we validate this? like download the template, and make sure it contains `profile` and `repository` if we are going to need it for initialization?

Reply:
I don't think so. We're validating the template URL in the API call now, and should be catching failed API calls anyway. If it lacks the initialize key (which is perfectly fine to do) then we should just skip initialization.

Reply:
I guess the thing to do here is make sure we're validating the form entry by submitting the newly-chosen template URL to `/setup/options` API?

Reply:
The yaml now gets error checked using the `setup/options` api.